neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+        J_mat = np.zeros_like(self.connectivity)
+        for col, mass_from in enumerate(self.masses):
+            # taus are constant per col and depends only on "from" mass
+            tau_mat[:, col] = mass_from.params[f"tau_syn_{mass_from.mass_type.lower()}"]
+            # Js are specific: take J from "to" mass but of type "from" mass
+            for row, mass_to in enumerate(self.masses):
+                J_mat[row, col] = mass_to.params[f"J_{mass_from.mass_type.lower()}_max"]
+
+        # multiplication with tau makes the increase of synaptic activity
+        # subject to a single input spike invariant to tau and division by J
+        # ensures that mu = J*s will result in a PSP of exactly c for a single
+        # spike
+        self.connectivity = (self.connectivity * tau_mat) / np.abs(J_mat)
+
+    def __init__(
+        self,
+        exc_params=None,
+        inh_params=None,
+        exc_lin_nonlin_cascade_filename=None,
+        inh_lin_nonlin_cascade_filename=None,
+        connectivity=DEFAULT_ADEX_NODE_CONNECTIVITY,
+        delays=DEFAULT_ADEX_NODE_DELAYS,
+        exc_seed=None,
+        inh_seed=None,
+    ):
+        """
+        :param exc_params: parameters for the excitatory mass
+        :type exc_params: dict|None
+        :param inh_params: parameters for the inhibitory mass
+        :type inh_params: dict|None
+        :param exc_lin_nonlin_cascade_filename: filename for precomputed
+            linear-nonlinear cascade for excitatory AdEx mass, if None, will
+            look for it in this directory
+        :type exc_lin_nonlin_cascade_filename: str|None
+        :param inh_lin_nonlin_cascade_filename: filename for precomputed
+            linear-nonlinear cascade for inhibitory AdEx mass, if None, will
+            look for it in this directory
+        :type inh_lin_nonlin_cascade_filename: str|None
+        :param connectivity: local connectivity matrix
+        :type connectivity: np.ndarray
+        :param delays: local delay matrix
+        :type delays: np.ndarray
+        :param exc_seed: seed for random number generator for the excitatory
+            mass
+        :type exc_seed: int|None
+        :param inh_seed: seed for random number generator for the inhibitory
+            mass
+        :type inh_seed: int|None
+        """
+        excitatory_mass = ExcitatoryAdExMass(
+            params=exc_params, lin_nonlin_cascade_filename=exc_lin_nonlin_cascade_filename, seed=exc_seed
+        )
+        excitatory_mass.index = 0
+        inhibitory_mass = InhibitoryAdExMass(
+            params=inh_params, lin_nonlin_cascade_filename=inh_lin_nonlin_cascade_filename, seed=inh_seed
+        )
+        inhibitory_mass.index = 1
+        super().__init__(
+            neural_masses=[excitatory_mass, inhibitory_mass], local_connectivity=connectivity, local_delays=delays,
+        )
+        self._rescale_connectivity()
+
+    def update_params(self, params_dict):
```

Comment:
as with individual masses - when updating parameters we need to check whether the connectivity itself is changed and if so - it has to be rescaled again